tasks/test.rake

```diff
@@ -42,9 +42,51 @@ task 'test:spec' do
 end
 
 ##
-## feature / integration tests
+## integration / smoke tests
 ##
 
+desc 'Executes smoke tests for a single gem'
+rule /^test:smoke:.+$/ do |task|
+  dir = "gems/#{task.name.split(':').last}/features"
+  tags = "-t '@smoke'"
+  ENV.fetch('AWS_SMOKE_TEST_SKIP_TAGS', '').split(',').each do |tag|
+    tags += " -t 'not @#{tag}'"
+  end
+  sh("bundle exec cucumber --retry 3 #{tags} -r #{dir} #{dir} --publish-quiet")
+end
+
+desc 'Executes all smoke tests'
+rule 'test:smoke' do
+  failures = []
+  Dir.glob('gems/*/features').each do |dir|
+    gem_name = dir.match(%r{gems/(.*)/features})[1]
+    sh("bundle exec rake test:smoke:#{gem_name}") do |ok, _|
+      failures << File.basename(File.dirname(dir)) unless ok
+    end
+  end
+  abort('one or more test suites failed: %s' % [failures.join(', ')]) unless failures.empty?
+end
+
+desc 'Executes integration tests for a single gem'
+rule /^test:integration:.+$/ do |task|
+  dir = "gems/#{task.name.split(':').last}/features"
+  tags = "-t 'not @smoke' -t 'not @veryslow'"
+  sh("bundle exec cucumber --retry 3 #{tags} -r #{dir} #{dir} --publish-quiet")
+end
+
+desc 'Executes all integration tests'
+task 'test:integration' do
+  failures = []
+  Dir.glob('gems/*/features').each do |dir|
+    gem_name = dir.match(%r{gems/(.*)/features})[1]
+    sh("bundle exec rake test:integration:#{gem_name}") do |ok, _|
```

Comment:
I'm wondering if we should change the format used when executing all of the tests.... It tends to produce so much output that it can be hard to search for the failures.

Maybe we shoudl run with `--format summary` or something like that? 

Or - long term, most test runners (eg: codebuild) have some junit integration and give nice displays - maybe we should be outputing junit format.  

Nothing we need to do now....

Reply:
I don't think we should do summary because it relies on color and some build logs like in ToD do not support that. I think the current format we have is fine and I think with the other suggestion you had, it is a lot more manageable.